src/pyodide/internal/snapshot.ts

```diff
@@ -26,8 +26,7 @@ let LOADED_BASELINE_SNAPSHOT: number;
 
 const TOP_LEVEL_SNAPSHOT =
   ArtifactBundler.isEwValidating() || SHOULD_SNAPSHOT_TO_DISK;
-const SHOULD_UPLOAD_SNAPSHOT =
-  ArtifactBundler.isEnabled() || TOP_LEVEL_SNAPSHOT;
+const SHOULD_UPLOAD_SNAPSHOT = TOP_LEVEL_SNAPSHOT;
```

Comment:
Rather than having two redundant names for this, why not combine them? I think `SHOULD_UPLOAD_SNAPSHOT` is probably the better name.